kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/AbstractRefinableView.java

```diff
@@ -223,4 +223,41 @@ public URI getUri() {
     }
     return builder.build();
   }
+
+  @Override
+  public void signalReady() {
+    Pair<DatasetRepository, Map<String, String>> pair =
+        Registration.lookupDatasetUri(URI.create(getUri().getRawSchemeSpecificPart()));
+    DatasetRepository repo = pair.first();
+
+    repo.signalReady(dataset.getNamespace(), dataset.getName(), getConstraints());
+  }
+
+  @Override
+  public boolean isReady() {
+    Pair<DatasetRepository, Map<String, String>> pair =
+        Registration.lookupDatasetUri(URI.create(getUri().getRawSchemeSpecificPart()));
+    DatasetRepository repo = pair.first();
```

Comment:
@rdblue - I presume your question here is why involve the MetadataProvider at all?  I think Ben's primary reason for involving the DatasetRepository was for access to the MetadataProvider.  I think the motivation for involving he MetadataProvider was to place the signals in a subdirectory of ".metadata".

I imagine instead they could just be placed in a ".signals" directory that sits on its own under the dataset's root directory.  If that were the case that would keep the DatasetRepository and MetadataProvider out of the picture.

Is that what you were thinking?